cmake/MacroGuardTest.cmake

```
@@ -0,0 +1,107 @@
+# PROJECT_NAME: bsoncxx or mongocxx.
+# GUARDED_MACROS: list of macros that should be guarded by prelude/postlude headers.
+# INCLUDE_FILTERS: list of regex filters for headers to be added to the tests (must be relative to PROJECT_SOURCE_DIR).
+# EXCLUDE_FILTERS: list of regex filters for headers to be excluded (applied after INCLUDE_FILTERS).
+# PROJECT_TEST_PROPERTIES: the (bsoncxx|mongocxx)_test_properties target to be linked with.
+function(add_macro_guard_test)
+    set(opt_args "")
+    set(single_args "PROJECT_NAME;PROJECT_TEST_PROPERTIES")
+    set(multi_args "GUARDED_MACROS;INCLUDE_FILTERS;EXCLUDE_FILTERS")
```

```suggestion
    set(multi_args "GUARDED_MACROS;INCLUDE_PATTERNS;EXCLUDE_REGEX")
```

Rename suggested since they accept different syntaxes